satellite/audit/reservoirchore.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/metainfo"
+)
+
+// ReservoirChore populates reservoirs and the audit queue.
+type ReservoirChore struct {
+	log    *zap.Logger
+	config Config
+	rand   *rand.Rand
+
+	service *Service2
+
+	MetainfoLoop *metainfo.Loop
+	Loop         sync2.Cycle
+	Interval     time.Duration
+}
+
+// NewReservoirChore instantiates ReservoirChore.
+func NewReservoirChore(log *zap.Logger, service *Service2, metaLoop *metainfo.Loop, config Config) *ReservoirChore {
+	return &ReservoirChore{
+		log:    log,
+		config: config,
+		rand:   rand.New(rand.NewSource(time.Now().Unix())),
+
+		service: service,
+
+		MetainfoLoop: metaLoop,
+		Loop:         *sync2.NewCycle(config.QueueInterval),
+		Interval:     config.ChoreInterval,
+	}
+}
+
+// Run starts the reservoir chore
+func (chore *ReservoirChore) Run(ctx context.Context) error {
+	return chore.Loop.Run(ctx, func(ctx context.Context) (err error) {
+		defer mon.Task()(&ctx)(&err)
+
+		select {
+		case <-ctx.Done():
```

Comment:
There's no need to check `ctx.Done` here because `chore.Loop.Run` already handles it.